### PR TITLE
Fix SearchFragment not restoring properly

### LINKED ISSUE #2561: Search - no turning back to other results
### Describe the bug

Hi,
since the new 0.15.4 version it is no longer possible to search something and return to other search results.
You are forced to enter a search term again. Tested on my 2 devices.
In version 0.15.3 the search went without any problems.

greetz

### Logs

_No response_

### Application version

0.15.4

### Where did you install the app from?

Amazon Appstore

### Device information

Amazon FireTV Stick 4k & FireTV Stick 4k MAX

### Android version

Android 7.1 - 6.2.9.5  /  Android 9 - 7.6.2.4

### Jellyfin server version

10.8.9

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
**Changes**
- Fix an issue where the behavior of SearchFragment updating the child fragment in onViewCreated caused state restoration to fail.

**Issues**

Closes #2561

### CODE REVIEW

**Comment by DavidFair** [comment]
I could reproduce this on `master` and can confirm this fixes the issue for me

(OT: I'm not sure what the repo policy is, can non maintainers approve a PR in addition to a maintainer or would you prefer us to only comment?)

**Review by crobibero — APPROVED** [review]

**Comment by nielsvanvelzen** [comment]
> I'm not sure what the repo policy is, can non maintainers approve a PR in addition to a maintainer or would you prefer us to only comment?

Both are fine, only approvals from team members count toward the "require 1 approval" merge condition we have set, but as repository admin I can always overrule it.

**Pull request merged**
